sparse_merkle_tree.js

```diff
@@ -68,12 +69,36 @@ class SparseMerkleTree {
     }
 
     async verify(root, queries, proof) {
-        return new Promise((resolve, reject) => {
-            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (err, result) => {
-                if (err) {
-                    reject(err);
-                    return;
+        return new Promise((resolve) => {
+            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (_, result) => {
```

Comment:
I think this has error case, so it should handle the error as the original code?